### PR TITLE
refactor(shared-config): extract chain + token metadata to shared package

### PR DESCRIPTION
**Pull request by chapati23**
## Summary

- Deletes `metrics-bridge/src/config.ts`'s hand-curated `POOL_PAIR_LABELS`, `CHAIN_NAMES`, `BLOCK_EXPLORER_BASE_URLS` — derives labels from `{chainId, token0, token1}` via shared helpers instead. Fixes the drift pattern that triggered #209 (Monad alerts rendered raw `143-0x93e1…`).
- Grows `shared-config/` from a 2-JSON package into a TS workspace package with `chains`, `tokens`, `format` exports consumed by `metrics-bridge` and `ui-dashboard`.
- `indexer-envio/src/feeToken.ts` intentionally NOT migrated — Envio may build outside the pnpm workspace (see `indexer-envio/src/contractAddresses.ts:14-18`). Left as a deliberate mirror + drift-protection test asserting `Mock*` exclusion + `decimals` required invariants still hold (security gate from #174).

## What changed

**shared-config (new):**
- `chain-metadata.json` — `chainId → {slug, label, explorerBaseUrl}`
- `src/chains.ts` — `chainSlug`, `chainLabel`, `hasChain`, `explorerBaseUrl`, `explorerAddressUrl`, `explorerTxUrl`
- `src/tokens.ts` — `tokenSymbol`, `poolName`, `chainTokenSymbols`, `chainAddressLabels`, `contractEntries` (raw walker for future indexer migration)
- `src/format.ts` — `poolIdAddress`, `shortAddress`
- `tsconfig.json`, `eslint.config.mjs`, `__tests__/` (vitest)

**metrics-bridge:**
- `config.ts` trimmed to env config only
- `metrics.ts` swaps imports + call sites; `warnIfUnknown` uses `hasChain` predicate (no more `chainSlug === String(chainId)` sentinel string-compare)
- `Dockerfile` copies `shared-config/` in both stages; prod stage uses `--ignore-scripts`
- `test/config.test.ts` deleted — coverage migrated to shared-config

**ui-dashboard:**
- `src/lib/networks.ts:buildNetworkMaps` delegates to shared `chainTokenSymbols` + `chainAddressLabels`
- Explorer URL defaults come from shared `chain-metadata.json`; env overrides + per-network `addressLabels` overrides preserved

**CI + build:**
- Root `postinstall` builds `shared-config/dist` so downstream typecheck sees `.d.ts` on fresh CI/Vercel installs
- New `shared` CI job; `bridge` filter + `metrics-bridge.yml` deploy paths include `shared-config/**`
- `.gcloudignore` no longer excludes `shared-config/` (Cloud Build needs it)

**AGENTS.md:** root + per-package docs updated to steer future sessions at the shared helpers.

## Regression coverage

- `shared-config/__tests__/fixtures/known-pools.json` captures the 11 pool entries from the old `POOL_PAIR_LABELS` map. `poolName(chainId, token0, token1) === expectedLabel` asserted for each before deletion.
- `indexer-envio/test/feeTokenSharedConfigSync.test.ts` asserts the fee-token allowlist keeps `Mock*` excluded + `decimals` required (the #174 security invariants).
- New coverage: `hasChain`, `poolName` single-null legs, `chainTokenSymbols` Mock retention, `chainTokenSymbols` non-token filter, `warnIfUnknown` one-shot dedup, partial fallback (known chain + unknown token), nullable `token0`/`token1` column.
- Total: 61 shared-config + 39 metrics-bridge + 1117 ui-dashboard + 303 indexer-envio = **1,520 tests pass**.

## Test plan

- [x] `pnpm -r typecheck && pnpm -r lint` — clean
- [x] `pnpm --filter @mento-protocol/monitoring-config test` — 61 passing
- [x] `pnpm --filter @mento-protocol/metrics-bridge test` — 39 passing (incl. new fallback/dedup/null-token tests)
- [x] `pnpm --filter @mento-protocol/ui-dashboard test` — 1117 passing
- [x] `pnpm --filter @mento-protocol/indexer-envio` tests — 303 passing (incl. 2 new drift-protection)
- [x] `./tools/trunk fmt --all && ./tools/trunk check --all` — clean
- [ ] CI `shared` / `ui` / `bridge` jobs green on this PR
- [ ] Spot-check Cloud Build (metrics-bridge deploy dry-run — happens automatically on merge via `.github/workflows/metrics-bridge.yml`)

## Review history

Ran `/review` (7 specialists) + `/codex-review` before opening. Fixed all confirmed blockers: `dist/` gitignored + no postinstall → downstream CI failure; Dockerfile missing `shared-config/` context; `.gcloudignore` exclusion; bridge CI filter missing `shared-config/**`. Addressed architecture warnings (`hasChain` predicate), coverage gaps (LARP + tests specialists both flagged missing Mock retention + partial-fallback cases), comment bloat (slop).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by chapati23** [comment]
@claude review

**Comment by chapati23** [comment]
cursor review

**Comment by chapati23** [comment]
@cursor[bot] thanks — acknowledged ✅ no findings to act on.

**Comment by chapati23** [comment]
@claude[bot] thanks for the review. Addressed four of the observations in c3e846b:

- **`canonicalTokenSymbol` empty-string guard** — added `name.length > 5` check so a hypothetical entry named exactly "Spoke" can't canonicalize to an empty symbol.
- **`warnIfUnknown` token identification** — warning now includes `token0=0x… , token1=0x…` when pair derivation fails, so on-call can jump straight to `@mento-protocol/contracts` without a Hasura round-trip.
- **Dockerfile redundant COPY** — folded the prod-stage `COPY shared-config/package.json` into the single `COPY shared-config/*.json` (the glob already covers it).
- **`shared` CI job Codecov upload** — now uploads coverage under the `monitoring-config` flag, matching the ui/indexer/bridge jobs.

Also added a `poolName(USDm, USDm) → "USDm/USDm"` test to document the same-token edge case for the USDm-last ordering.

Skipping:
- **`shortAddress` 12-char threshold** — you noted it's cosmetic; the current 4+4-nibble compromise is deliberate for fitting in Slack alert titles, so keeping as-is.
- **`networks.ts` triple explorer fallback** — the `env ?? shared ?? hardcoded` floor is a deliberate paranoia layer for misconfigured runtimes; worth keeping even though the hardcoded values mirror the JSON.

**Comment by chapati23** [comment]
@claude review

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "private": true,
   "packageManager": "pnpm@10.30.3",
   "scripts": {
+    "postinstall": "pnpm --filter @mento-protocol/monitoring-config build",
```
[the remark this answers is not in the document]
Comment (chapati23) [reply package.json:6]: Fixed in 16711d1 — good catch, confirmed real. Reproduced locally: `pnpm install --filter @mento-protocol/metrics-bridge` leaves `shared-config/node_modules/@mento-protocol/` empty, so shared-config's postinstall `tsc` can't find `@mento-protocol/contracts/contracts.json`. Added trailing `...` to both builder and prod `pnpm install` filter arguments in `metrics-bridge/Dockerfile` so transitive workspace deps get installed.

File: metrics-bridge/Dockerfile [hunk 2]
```diff
@@ -16,9 +20,17 @@ RUN addgroup -S bridge && adduser -S bridge -G bridge
...
 COPY metrics-bridge/package.json metrics-bridge/
-RUN corepack enable && pnpm install --prod --frozen-lockfile --filter @mento-protocol/metrics-bridge
+# Copy shared-config's package.json + runtime JSON exports (the `exports` map
+# points at `./*.json` directly). Covers both the install step below AND the
+# runtime JSON resolution in the final image.
+COPY shared-config/*.json shared-config/
+# --ignore-scripts skips the root `postinstall` (which runs `tsc` on shared-config)
+# because the dist is copied from the builder below and the prod install has no
+# devDeps. The workspace symlink still resolves at runtime.
+RUN corepack enable && pnpm install --prod --frozen-lockfile --ignore-scripts --filter @mento-protocol/metrics-bridge
```
[the remark this answers is not in the document]
Comment (chapati23) [reply metrics-bridge/Dockerfile]: Fixed in 16711d1 — same root cause as the sibling finding on `package.json:6`. Runtime `shared-config/dist/tokens.js` does `require('@mento-protocol/contracts/contracts.json')`, which would crash the image at startup without the trailing `...` filter. Applied the fix to both Dockerfile install commands and verified locally that `shared-config/node_modules/@mento-protocol/contracts` is now present.

**Comment by chapati23** [comment]
@claude[bot] thanks, but the Codex findings were **not** false positives — I reproduced them locally before pushing the fix:

- **pnpm filter `...` semantics:** trailing `pattern...` means **dependencies** (packages pattern depends on); leading `...pattern` means **dependents** (packages that depend on pattern). See [pnpm --filter docs](https://pnpm.io/filtering#--filter-package_name). The review got the arrow reversed.
- **Empirical repro:** I copied the workspace metadata to a scratch dir and ran `pnpm install --frozen-lockfile --filter @mento-protocol/metrics-bridge --ignore-scripts`. Result: `shared-config/node_modules/@mento-protocol/` is **empty** — no `contracts` symlink. Re-running with `--filter '@mento-protocol/metrics-bridge...'` restored it. That matches exactly what Codex described.
- **Vercel preview ≠ Docker build:** Vercel builds `ui-dashboard` (root-install, no `--filter`). The Cloud Run bridge image goes through `metrics-bridge/Dockerfile` via `cloudbuild.yaml` on push-to-main. PR CI does not exercise that path, so the preview passing isn't evidence either way.

Also fixed your minor observation about the misleading 'its postinstall' comment in 02a08bc — shared-config has no postinstall of its own; it's the root workspace postinstall that runs tsc. Clarified accordingly.
